File: sample/src/main/java/io/jjuni/sample/lifecycledetect/MainActivity.java
package io.jjuni.sample.lifecycledetect;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import io.jjuni.library.lifecycledetect.LifecycleDetectActivity;

public class MainActivity extends LifecycleDetectActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }


}

File: sample/src/main/java/io/jjuni/sample/lifecycledetect/MainActivity.kt
package io.jjuni.sample.lifecycledetect

import android.annotation.SuppressLint
import android.os.Bundle
import android.os.PersistableBundle
import androidx.lifecycle.ViewModelProviders
import io.jjuni.library.lifecycledetect.LifecycleDetectActivity

class MainActivity : LifecycleDetectActivity() {

    private val viewModel: MainViewModel by lazy {
        ViewModelProviders.of(this).get(MainViewModel::class.java)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        register(viewModel)
    }

}
